### ISSUE TITLE
Defaults are not honored?

### ISSUE THREAD

**Issue by nmenon** [ulfalizer/Kconfiglib#57]
Consider the following Kconfig file.
[Kconfig.txt](https://github.com/ulfalizer/Kconfiglib/files/2544862/Kconfig.txt)

As per https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt, 

> - default value: "default" <expr> ["if" <expr>]
>   A config option can have any number of default values. If multiple
>   default values are visible, only the first defined one is active.
>   Default values are not limited to the menu entry where they are
>   defined. This means the default can be defined somewhere else or be
>   overridden by an earlier definition.
>   The default value is only assigned to the config symbol if no other
>   value was set by the user (via the input prompt above). If an input
>   prompt is visible the default value is presented to the user and can
>   be overridden by him.
>   Optionally, dependencies only for this default value can be added with
>   "if"

Now with a .config file such as:
[config-orig.txt](https://github.com/ulfalizer/Kconfiglib/files/2544863/config-orig.txt)

running oldconfig, ends up with the following result:
[config-gen.txt](https://github.com/ulfalizer/Kconfiglib/files/2544864/config-gen.txt)

**Comment by ulfalizer** [comment]
Yep, only symbols with prompts can be assigned by the user (can be "visible", in Kconfig speak).

Those are the only symbols that show up in the `menuconfig` interface too (unless you turn on show-all mode in the terminal menuconfig).

The reason other symbols still get written out is that `.config` files can be used as input to Make, which needs to see all the values.
